program-runtime/src/invoke_context.rs

```diff
@@ -381,6 +364,20 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
                 Err(InstructionError::MissingAccount)
             };
             instruction.visit_each_account(&mut work)?;
+        } else {
```

Comment:
This rearrangement looks good on inspection but just to ask out loud.. this is safe to reorder because the `visit_each_account` call is infallible right?

Reply:
This should not reorder anything because it is in the `else` block.

Before we had: `ReentrancyNotAllowed` before `MissingAccount`
Now we have: Either `ReentrancyNotAllowed` or `MissingAccount`

Reply:
Ah yes, you're correct, thanks